sdk/java/dagger-java-annotation-processor/src/test/resources/io/dagger/java/module/DaggerJava.java

```diff
@@ -37,7 +35,9 @@ public Container containerEcho(@Default("Hello Dagger") String stringArg) {
    * @return Standard output of the grep command
    */
   @Function
-  public String grepDir(Directory directoryArg, @Nullable String pattern)
+  public String grepDir(
+      @DefaultPath("sdk/java") @Ignore({"**", "!*.java"}) Directory directoryArg,
+      @Nullable String pattern)
```

Comment:
Could `java.util.Optional` be used instead of `@Nullable`?

Like:
```diff
-public String grepDir(Directory directoryArg, @Nullable String pattern)
+public String grepDir(Directory directoryArg, Optional<String> pattern)
```
Or:
```diff
-public String nullableDefault(@Nullable @Default("Foo") String stringArg) {
+public String nullableDefault(@Default("Foo") Optional<String> stringArg) {
```

Reply:
That's a good question. But it makes things a little bit more complexe, especially as `Optional` is not a serializable type. This doesn't mean we can't do it, but it requires more type manipulation.

I'll investigate a bit more, but in a different PR so this doesn't block this one that is on the `@DefaultPath`/`@Ignore`.

Reply:
@helderco agree, I think we should try to stick as closely as possible to the language's idioms.
@eunomie This will require some minor changes in how module calls are handled in the entry point code, but nothing too complex at first glance. That said, we can definitely address this in another PR.

Reply:
Of course, wasn't suggesting it would be in this PR 👍